Autocoders/Python/src/fprime_ac/generators/templates/serialize/publicSerialH.tmpl

```diff
@@ -41,20 +41,20 @@ public:
 
     void set($args_proto); //!< set values
 
-#for name,type,size,format,comment,typeinfo in $members:
-#if $size == None:
+#for name,type,array_size,size,format,comment,typeinfo in $members:
+#if $array_size == None and not ($typeinfo == 'string' or $typeinfo == 'extern'):
     ${type} get${name}() const; //!< get member $name
-#else if $typeinfo == 'string' or $typeinfo == 'extern':
+#else if ($typeinfo == 'string' or $typeinfo == 'extern') and array_size == None:
     const ${type}& get${name}() const; //!< get member $name
```

Comment:
Why can't we return a reference to the array type?

Reply:
If there's a struct member array, we return a pointer to it in line 50. This is also what the previous code did.

The previous code said if the type is string or extern (i.e., serializable), assume no array and return the single element as a reference. The new code relaxes this assumption. It says if we see string or serializable and there's no struct member array, then return the single element as a reference. Otherwise fall through and return the struct member array as a pointer.

I think it's correct.

Reply:
Except the new logic may be unnecessarily complicated. I think the condition in line 47 should just be 'if array_size == None'.